llvm-project/Dockerfile

```diff
@@ -0,0 +1,133 @@
+FROM alpine:edge AS source
+RUN wget --no-clobber \
+  https://github.com/llvm/llvm-project/releases/download/llvmorg-14.0.1/llvm-project-14.0.1.src.tar.xz
+
+# Stage one
+FROM alpine:edge AS stage_one
+
+# Install the stage zero compiler; the prebuilt clang from alpine.
+# _Unwind_Resume issues with llvm-libunwind? Cmake can't find it either?
+# TODO: can we build stage1 with llvm binutils? llvm13 alpine package contains.
+#RUN apk add --no-cache clang lld compiler-rt musl-dev
+RUN apk add --no-cache clang lld musl-dev
+
+COPY --from=source llvm-project-14.0.1.src.tar.xz /
+RUN tar xf llvm-project-14.0.1.src.tar.xz && \
+  mv llvm-project-14.0.1.src llvm-project && \
+  mkdir llvm-project/llvm/build
+
+# libcxxabi depends on zlib-dev!
+# compiler-rt depends on libexecinfo-dev! (unless gwp-asan is disabled)
+# TODO: libxml2?
+# llvm cmake throws lots of warnings without git
+RUN apk add --no-cache cmake ninja python3 zlib-dev git
+
+# This seems like a mistake that clang++ in alpine cannot compile C++ hello
+# world.
+# TODO: this also pulls in gnu binutils...
+RUN apk add --no-cache g++ && \
```

Comment:
```
$ docker run -it alpine:edge ash
/ # apk add clang
/ # clang++ x.cpp
x.cpp:1:10: fatal error: 'iostream' file not found
#include <iostream>
         ^~~~~~~~~~
1 error generated.
/ # apk add g++
/ # clang++ x.cpp
/ #
```

I think the content list on alpine's site is broken for g++; it's not listing any files.
https://pkgs.alpinelinux.org/contents?branch=edge&name=g++&arch=x86_64&repo=main
But:
```
apk info -L g++| grep iostream
usr/include/c++/11.2.1/iostream
```
so we need to install g++ on alpine to get the libstdc++ headers (TBH; that looks like a mistake to me, I think the headers should be distributed with libstdc++ package).

> a cleaner way to do this might be looking at the contents of the gcc and g++ packages so that we are explicitly removing every binary from the packages?
```
/ # apk info -L g++ gcc| grep "^usr/bin"
usr/bin/c++
usr/bin/g++
usr/bin/x86_64-alpine-linux-musl-c++
usr/bin/x86_64-alpine-linux-musl-g++
usr/bin/c89
usr/bin/c99
usr/bin/cc
usr/bin/cpp
usr/bin/gcc
usr/bin/gcc-ar
usr/bin/gcc-nm
usr/bin/gcc-ranlib
usr/bin/gcov
usr/bin/gcov-dump
usr/bin/gcov-tool
usr/bin/lto-dump
usr/bin/x86_64-alpine-linux-musl-gcc
usr/bin/x86_64-alpine-linux-musl-gcc-11.2.1
usr/bin/x86_64-alpine-linux-musl-gcc-ar
usr/bin/x86_64-alpine-linux-musl-gcc-nm
usr/bin/x86_64-alpine-linux-musl-gcc-ranlib
/ # apk info -L g++ gcc| grep "^usr/bin" | xargs rm
/ # clang x.cpp
/ #
```
:)

Reply:
done. PTAL